Sources/FaunaDB/Expr.swift

```diff
@@ -1,5 +1,47 @@
 import Foundation
 
+/**
+    `Expr` is the top level of the query language expression tree.
+
+    For convenience, some native types are considered valid expressions:
+    `String`, `Int`, `Double`, `Bool`, `Optional`, and `Date`.
+
+    ## Motivation
+
+    The FaunaDB query language is a non-typed expression based language.
+    You can think about it as an actual programming language.
+
+    In order to express the whole power and flexibility of the query language,
+    we need to express our DSL in a way that allows the same combinations as a
```

Comment:
I've found this really tricky to explain even to myself sometimes. My first reaction when I started to work with the driver back in the day when I was doing the Java driver was to try to create a type aware DSL. It took me a while to realise that trying to make the DSL typed would restrict function composition.

The original developers of the Swift Driver had the same hunch.

I'll think about this more carefully.